Framework/Algorithms/src/CalculateMultipleScattering.cpp

```diff
@@ -752,5 +783,34 @@ void CalculateMultipleScattering::interpolateFromSparse(MatrixWorkspace &targetW
   PARALLEL_CHECK_INTERUPT_REGION
 }
 
+void CalculateMultipleScattering::correctForWorkspaceNameClash(std::string &wsName) {
+  bool noClash(false);
+
+  for (int i = 0; !noClash; ++i) {
+    std::string wsIndex; // dont use an index if there is no other
+                         // workspace
+    if (i > 0) {
+      wsIndex = "_" + std::to_string(i);
+    }
+
+    bool wsExists = AnalysisDataService::Instance().doesExist(wsName + wsIndex);
+    if (!wsExists) {
+      wsName += wsIndex;
+      noClash = true;
+    }
+  }
+}
+
+/**
+ * Set the name on a workspace, adjusting for potential clashes in the ADS.
+ * Used to set the names on the output workspace group members. N
+ * @param ws The ws to set the name on
+ * @param wsName The name to set on the workspace
+ */
+void CalculateMultipleScattering::setWorkspaceName(const API::MatrixWorkspace_sptr &ws, std::string wsName) {
```

Comment:
If someone runs CalculateMultipleScattering twice, will this result in the numbers incrementing from `name_n` to `name_2n` or overwrite it?

I ask as this differs from the normal behaviour of Mantid Algorithms which is to overwrite. So this becomes the exception to the rule when setting the output name to something that already exists? Equally, I may be misreading this a bit...

Reply:
I made this change to try and deal with the situation where someone runs it twice with OutputWorkspace set to different values eg "MuscatResultsOne" and "MuscatResultsTwo". As a result of this change they will end up with the following workspaces where the unindented ones are group workspaces:

MuscatResultsOne
- Scatter_1_No_Abs
- Scatter_1
- Scatter_2

MuscatResultsTwo
- Scatter_1_No_Abs_1
- Scatter_1_1
- Scatter_2_1

Before this change you'd end up with two group workspaces containing the same child workspaces. These get listed twice in the ADS so it was a bit confusing.

I think you have a good point though. I seem to have broken the more simple case where someone just runs the algorithm twice with the same settings. This should overwrite the previous outputs. I should probably add a line to try and delete the output group workspace named in the "OutputWorkspace" parameter from the ADS (including its children) before writing any of the new outputs.

The method correctForWorkspaceNameClash is copied from LoadNexusProcessed::correctForWorkspaceNameClash so hopefully the approach for avoiding the name clash when it's actually needed is fairly standard